save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/AgentsController.kt

```diff
@@ -57,9 +56,22 @@ class AgentsController(private val agentStatusRepository: AgentStatusRepository,
      */
     @GetMapping("/getAgentsStatusesForSameExecution")
     @Transactional
-    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> = agentRepository
-        .findByExecutionIdOfContainerId(agentId)
-        .map {
-            agentStatusRepository.findTopByAgentContainerIdOrderByTimeDesc(it.containerId)?.toDto()
-        }
+    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> = agentRepository.findAll { root, cq, cb ->
+        cb.equal(root.get<Execution>("execution"), getAgentByContainerId(agentId).execution)
```

Comment:
Is `"execution"` here a name of column or a name of a class property? In latter case, is there any overload for this method for property reference?

Reply:
Also, is there any difference if we compare entire `Execution`s or only their IDs?

Reply:
No difference. When building a request, it still uses the ID.